src/features/wallet/send/SendView.tsx

```diff
@@ -420,33 +372,43 @@ const SendView = ({ scanResult, sendType, hotspot, isSeller }: Props) => {
     }
   }
 
-  const constructTxn = async () => {
+  const constructTxns = async () => {
     if (type === 'payment') {
-      return makePaymentTxn(balanceAmount.integerBalance, address, getNonce())
+      return Promise.all(
+        sendTransfers.map(async ({ address, balanceAmount }) => {
+          return makePaymentTxn(
+            balanceAmount.integerBalance,
+            address,
+            getNonce(),
+          )
+        }),
+      )
     }
 
     if (type === 'dc_burn') {
-      return makeBurnTxn(
-        balanceAmount.integerBalance,
-        address,
-        getNonce(),
-        memo,
+      return Promise.all(
+        sendTransfers.map(async ({ address, balanceAmount, memo }) => {
+          return makeBurnTxn(
+            balanceAmount.integerBalance,
+            address,
+            getNonce(),
+            memo,
+          )
+        }),
       )
     }
 
     if (type === 'transfer') {
-      return isSeller ? handleSellerTransfer() : handleBuyerTransfer()
+      return [isSeller ? handleSellerTransfer() : handleBuyerTransfer()]
     }
 
     throw new Error('Unsupported transaction type')
   }
 
   const handleSubmit = async () => {
     try {
-      const txn = await constructTxn()
-      if (txn) {
-        await submitTxn(txn)
-      }
+      const txns = await constructTxns()
+      await Promise.all(txns.map(async (txn: any) => submitTxn(txn)))
```

Comment:
After addressing my other comments about this, we will need to change this back to just submitting one transaction.